Remove obsolete disabling of link check

avbentem opened the pull request:
Disabled “LMIC_setLinkCheckMode(0)” as meanwhile TTN supports that; see https://github.com/TheThingsNetwork/ttn/commit/13e3c6d

matthijskooijman commented:
Hey, missed the announcement about that getting implemented. For my understanding, is this implemented in the newest production backend only, or also on the staging backend (that is being phased out now)?

avbentem commented:
Good point, I'll try to figure that out :-)

(I noticed seeing downlinks in the new console, while not asking for them. That's for https://github.com/JaapBraam/LoRaWanGateway connecting to `router.eu.things.network`. Will try with the older domains, and/or will ask TTN.)

tftelkamp commented:
LoRaWAN ADRACKREQ is indeed implemented in the v2 (production) backend, not in staging.

Not sure though if it's a good idea to enable it by default. It really only make sense in combination with ADR, which is not yet implemented in TTN. Without ADR, eventually nodes all move in the direction of SF12, because if the downlink fails for whatever reason, the node will lower the data rate. ADR would be the mechanism to increase it again.

avbentem commented:
@tftelkamp, for future readers (please correct me if I'm wrong), as your comment got me wondering if link check would _ever_ be useful:

- Without ADR, apparently _"eventually nodes all move in the direction of SF12"_. Bummer.
- With ADR, a node still does not know if its (first) messages were received (although it would know that the join was successful), so could use link check to ensure something was received. (If its messages were not received, it would obviously also never get ADR commands.)

I'll change the pull request to only update the documentation, and leave it disabled.

avbentem commented:
@tftelkamp, while ADR was not implemented when I submitted the above, it seems it's now: TheThingsNetwork/ttn#265

So, how do you now feel about no longer disabling it? Maybe using `#if !defined(CFG_eu868)`, as maybe it's only for EU:
 
> **htdvisser** Feb 7th at 2:53 PM
> As we just updated all public routing servers to `v2.5.0`, I thought it would be nice to let you guys know what new features we added to the backend in the last month, so here's the summary :slightly_smiling_face:
>
> ...
> Adaptive Data Rate for Europe 863-870
> ...

(Copied [from Slack](https://thethingsnetwork.slack.com/archives/backend/p1486475631000974), but Slack's archive might have been purged by the time one clicks the link.)

tftelkamp commented:
On stati nodes it is good to enable ADR and LinkCheck, for mobile devices keep both turned off.

avbentem commented:
Some test today show that using the following will NOT make TTN send ADR:

```cpp
LMIC_setAdrMode(1);
LMIC_setLinkCheckMode(0);
```

Leaving out the latter makes ADR work just great.
